gson/src/main/java/com/google/gson/internal/UnsafeAllocator.java

```diff
@@ -101,4 +105,18 @@ public static UnsafeAllocator create() {
       }
     };
   }
+
+  /**
+   * Check if the class can be instantiated by unsafe allocator. If the instance has interface or abstract modifiers
+   * throw an {@link java.lang.UnsupportedOperationException}
+   * @param c instance of the class to be checked
+   */
+  private static void assertInstantiable(Class c) {
+    if (Modifier.isInterface(c.getModifiers())) {
```

Comment:
nit: can you extract a variable and keep `c.getModifiers()` ? It’s not necessarily a fast method on all VMs